### PR TITLE
feat: add zhipu embedding-3

### LINKED ISSUE #7096: Support for Zhipuai embedding-3
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

I want to use zhipuai embedding-3.

### 2. Additional context or comments

_No response_

### 3. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

### PR DESCRIPTION
**Pull request by orangeclk**
# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [x] Please open an issue before creating a PR or link to an existing issue
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Description

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue. Close issue syntax: `Fixes #<issue number>`, see [documentation](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) for more details.

Fixes #7096

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [ ] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# Testing Instructions

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x]  Apply retrieval testing in the knowledge tab
- [x]  Run pytest zhipuai
image
image
image

### CODE REVIEW

File: api/core/model_runtime/model_providers/zhipuai/text_embedding/embedding-3.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+model: embedding-3
+model_type: text-embedding
+model_properties:
+  context_size: 8192
```
Comment (crazywoola) [inline api/core/model_runtime/model_providers/zhipuai/text_embedding/embedding-3.yaml:4]:
```suggestion
  context_size: 2048
```

**Review by crazywoola — CHANGES_REQUESTED** [review]

File: api/core/model_runtime/model_providers/zhipuai/text_embedding/embedding-3.yaml (hunk 1 above)
Comment (orangeclk) [reply api/core/model_runtime/model_providers/zhipuai/text_embedding/embedding-3.yaml:4]: https://bigmodel.cn/dev/howuse/model
image

I found that the document of zhipu models has been updated. It shows that the context windows is 8k. The default demension is 2k. I'm not sure if I should also change the context window of embedding-2 in this PR. Will it be clearer to split it into two PRs?

Thank you.

Comment (crazywoola) [reply api/core/model_runtime/model_providers/zhipuai/text_embedding/embedding-3.yaml:4]: Yes, please update both context size in this PR. Thank you :)

**Review by crazywoola — APPROVED** [review]

File: api/core/model_runtime/model_providers/zhipuai/text_embedding/embedding-3.yaml (hunk 1 above)
Comment (orangeclk) [reply api/core/model_runtime/model_providers/zhipuai/text_embedding/embedding-3.yaml:4]: It has been pushed.

**Pull request merged**